src/thd_dbus_interface.cpp

```diff
@@ -580,6 +580,670 @@ gboolean thd_dbus_interface_get_sensor_temperature(PrefObject *obj, int index,
+		g_autofree gchar *path = NULL;
+		gint min_state;
+		gint max_state;
+		gint step;
+
+		g_variant_get(parameters, "(ssiii)", &cdev_name, &path, &min_state,
+			      &max_state, &step);
+
+		ret = thd_dbus_interface_update_cooling_device(obj,
+							       cdev_name, path, min_state, max_state,
+							       step, &error);
+		
+		if (!ret) {
+			g_dbus_method_invocation_return_gerror(invocation, error);
+			return;
+		}
+
+		g_dbus_method_invocation_return_value(invocation, NULL);
+		return;
+	}
+
+	g_set_error(&error,
+		    G_DBUS_ERROR,
+		    G_DBUS_ERROR_UNKNOWN_METHOD,
+		    "no such method %s",
+		    method_name);
+	g_dbus_method_invocation_return_gerror(invocation, error);
+}
+
+static GVariant *
+thd_dbus_handle_get_property(GDBusConnection  *connection,
+			     const gchar      *sender,
+			     const gchar      *object_path,
+			     const gchar      *interface_name,
+			     const gchar      *property_name,
+			     GError          **error,
+			     gpointer          user_data)
+{
+	return NULL;
+}
+
+static gboolean
+thd_dbus_handle_set_property(GDBusConnection  *connection,
+			     const gchar      *sender,
+			     const gchar      *object_path,
+			     const gchar      *interface_name,
+			     const gchar      *property_name,
+			     GVariant         *value,
+			     GError          **error,
+			     gpointer          user_data) {
+	return TRUE;
+}
+
+
+static void
+thd_dbus_on_bus_acquired(GDBusConnection *connection,
+			 const gchar     *name,
+			 gpointer         user_data) {
+	guint registration_id;
+	GDBusProxy *proxy_id = NULL;
+	GError *error;
+	GDBusNodeInfo *introspection_data = NULL;
+
+	if (user_data == NULL) {
+		thd_log_error("user_data is NULL\n");
+		return;
+	}
+
+	introspection_data = thd_dbus_load_introspection("src/thd_dbus_interface.xml",
+							 &error);
+	if (error != NULL) {
+		thd_log_error("Couldn't create introspection data: %s:\n",
+			      error->message);
+		return;
+	}
+
+	registration_id = g_dbus_connection_register_object(connection,
+							    "/org/freedesktop/thermald",
+							    introspection_data->interfaces[0],
+							    &interface_vtable,
+							    user_data,
+							    NULL,
+							    &error);
+
+	proxy_id = g_dbus_proxy_new_sync(connection,
+					 G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES,
+					 NULL,
+					 "org.freedesktop.DBus",
+					 "/org/freedesktop/DBus",
+					 "org.freedesktop.DBus",
+					 NULL,
+					 &error);
+	g_assert(registration_id > 0);
+	g_assert(proxy_id != NULL);
+}
+
+static void
+thd_dbus_on_name_acquired(GDBusConnection *connection,
+			  const gchar     *name,
+			  gpointer         user_data) {
+}
+
+static void
+thd_dbus_on_name_lost(GDBusConnection *connection,
+		      const gchar     *name,
+		      gpointer         user_data)
+{
+	g_warning("Lost the name %s\n", name);
+	exit(1);
+}
+
+
+// Set up Dbus server with GDBus
+int thd_dbus_server_init(gboolean (*exit_handler)(void)) {
+	GError *error = NULL;
+	PrefObject *value_obj;
+
```

Comment:
Maybe need to add this line?
thd_dbus_exit_callback = exit_handler;

@spandruvada , can you check if terminate dbus message still work?

In my case, intel_lpmd fails to terminate via dbus message without setting the proper exit handler.